registry/storage/cache/metrics/prom.go

```diff
@@ -0,0 +1,69 @@
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/docker/distribution"
+	prometheus "github.com/docker/distribution/metrics"
+	"github.com/docker/distribution/registry/storage/cache"
+	"github.com/docker/go-metrics"
+	"github.com/opencontainers/go-digest"
+)
+
+type prometheusCacheProvider struct {
+	cache.BlobDescriptorCacheProvider
+	latencyTimer metrics.LabeledTimer
+}
+
+func NewPrometheusCacheProvider(wrap cache.BlobDescriptorCacheProvider, name, help string) cache.BlobDescriptorCacheProvider {
```

Comment:
I wonder if we should document all the metrics exposed by registry under `docs/`. Need not be this PR.

Reply:
That sounds like a good exercise in which I would be interested in the result

Reply:
A single `/metrics` output should be sufficient to start with. I've seen other projects do that.